### PR TITLE
<refactor> Move output file naming into engine

### PR DESCRIPTION
**Pull request by roleyfoley**
Moves the output file suffix into the engine rather than createTemplate 

This is part of moving output definitions into the engine completely rather than half in the engine half in script

### CODE REVIEW

**Comment by roleyfoley** [comment]
@ml019 @RossMurr4y This is now ready for review

**Comment by RossMurr4y** [comment]
Looks good to me and working as expected with the Azure provider.

File: providers/aws/inputsources/mock/commandlineoption.ftl [hunk 1]
```diff
@@ -2,14 +2,17 @@
...
 [#macro aws_input_mock_commandlineoption_seed ]
+
     [@addCommandLineOption
         option={
             "Regions" : {
                 "Segment" : "ap-southeast-2",
-                "Product" : "ap-southeast-2",
                 "Account" : "ap-southeast-2"
+            },
+            "References" : {
```
Comment (ml019) [inline providers/aws/inputsources/mock/commandlineoption.ftl]: Is this test data?

Comment (roleyfoley) [reply providers/aws/inputsources/mock/commandlineoption.ftl]: Yep that's the mock data input

**Pull request merged**